src/mongo/db/storage/wiredtiger/encryption_keydb.h

```diff
@@ -46,22 +47,59 @@ namespace mongo {
 class EncryptionKeyDB
 {
 public:
-    EncryptionKeyDB(const std::string& path, const bool rotation = false)
-        : EncryptionKeyDB(false, path, rotation) {}
-    EncryptionKeyDB(const bool just_created, const std::string& path)
-        : EncryptionKeyDB(just_created, path, false) {}
     ~EncryptionKeyDB();
 
-    // tries to read master key from specified file
-    // then opens WT connection
-    // throws exceptions if something goes wrong
-    void init();
-
-    // during rotation copies data from provided instance
-    void clone(EncryptionKeyDB *old);
-
-    // write master key to the Vault (during rotation)
-    void store_masterkey();
+    /// @brief Creates a (possibly encrypted at rest) new encryption key database.
+    ///
+    /// @param path Path to the directory where the key db should be created;
+    ///     the path must point to an existing empty directory
+    /// @param kmipMasterKeyId In the case encryption is enabled and KMIP is used as
+    ///     a key management solution, this is the identifier of the database master key.
+    ///     If it is an empty string, then the function automatically creates a master key.
+    ///     and uploads it to the KMIP server specified in configuration.
+    ///     If KMIP is not used, the parameter is ignored.
+    ///
+    /// @returns Crated key database
+    ///
+    /// @throws std::runtime_error if
+    ///     1. the master key can't be read from or written to a KMIP or Vault server
+    ///     2. Vault token file can't be read
+    ///     3. can't create encryption key db at the specified path
+    static std::unique_ptr<EncryptionKeyDB> create(const std::string& path,
+                                                   const std::string& kmipMasterKeyId);
+
+    /// @brief Reads (possibly encrypted at rest) the encryption key database.
```

Comment:
Same: "possibly encrypted at rest" is misleading here.

Reply:
In fact it would be better to name this function 'open' because it does not "read" database it "opens" a handle to the existing database. (while 'create' creates db and opens a handle to it)